### DISCUSSION TITLE
Welcome to dynamoit Discussions!

### DISCUSSION THREAD

**Discussion by bykka** [bykka/dynamoit#11] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by prayforbirdies** [comment]
I have no idea why this isn't the number 1 project on github. Man this is fantastic, so thanks. 

A couple questions. 1) is there any plans for a csv export? Right now I just use python to convert so not a big deal just curious. 
2) Is there any way to execute a query or PartiQL query? I'd love to limit my results to a few attributes or columns.

Thanks again, my build failed 1 test I couldn't figure out but I skipped and was able to generate the jar. (attached) 
![Screenshot 2024-09-24 at 7 11 26 PM](https://github.com/user-attachments/assets/9c782008-a429-4074-b3e7-a2dfacf24207)

**Reply by bykka** [reply]
Hi, thank you for your interest in my project.
First of all to build the project you need Java 17. 

Now answers to your questions:
1. CSV export is not hard to add if we assume that the document structure is flat. Excel could be even better because it does not require escape double quotes (If some properties have a complex(tree) structure and default serialization for such properties will be in JSON format).
2. The project uses AWS SDK v1 for Java. This version does not support PartiQL. I'm trying to migrate it to version 2 but Api there completely different, plus some functionality has disappeared. But yes I thought to add PartiQL support.

**Comment by prayforbirdies** [comment]
Thanks for the reply! I've csv export sorted , takes a few seconds to convert the json. The fact that i can export everything is a massive help. So thanks!

Should I upgrade my Java and give it a shot? I'm on 17 but 
![Screenshot 2024-09-25 at 2 02 09 PM](https://github.com/user-attachments/assets/52d198e2-b862-4826-b75d-31f6f60401e4)

**Comment by prayforbirdies** [comment]
nevermind I got it to build with this version of 17!
![Screenshot 2024-09-25 at 2 15 05 PM](https://github.com/user-attachments/assets/630035aa-c12a-4bef-a8a0-e6b0fa292b3c)

**Reply by bykka** [reply]
You could simply download a release bundle https://github.com/bykka/dynamoit/releases

**Reply by prayforbirdies** [reply]
> You could simply download a release bundle https://github.com/bykka/dynamoit/releases

HAHAHAHAHAH I'm a moron. :-)

**Reply by prayforbirdies** [reply]
Is there something I'm missing though? It wont open on seqouia
![Screenshot 2024-09-25 at 3 25 49 PM](https://github.com/user-attachments/assets/1e857e7a-058c-4767-9d71-1b7ef845408a)

**Reply by bykka** [reply]
Yep, it seems like something has been altered. I created the build script a few years ago and everything was working fine at that time. Since I use Windows, I didn't notice that the Mac version is now broken. It's good to know - I'll create a new issue for this. Thanks.
